src/main/java/org/openstreetmap/atlas/checks/flag/CheckFlag.java

```diff
@@ -428,4 +437,63 @@ public String toString()
     {
         return String.format("[CheckFlag: %s, %s]", this.identifier, this.getInstructions());
     }
+
+    public JsonObject asGeoJsonFeature()
+    {
+        final JsonObject geometry = boundsGeoJsonGeometry();
+
+        final JsonObject properties = new JsonObject();
+        properties.addProperty("flag:type", CheckFlag.class.getSimpleName());
+        properties.addProperty("flag:id", getIdentifier());
+        properties.addProperty("flag:instructions", getInstructions());
+
+        // The legacy GeoJSON FeatureCollection doesn't actually provide this,
+        // but I figure this might be useful to know about if it's there...
+        if (challengeName != null)
+        {
+            properties.addProperty("flag:challenge", challengeName);
+        }
+
+        return GeoJsonUtils.feature(geometry, properties);
+    }
+
+    private JsonObject boundsGeoJsonGeometry()
+    {
+        final Iterator<FlaggedObject> iterator = flaggedObjects.iterator();
+        Rectangle bounds;
+
+        // Get the first bounds.
+        if (iterator.hasNext())
+        {
+            bounds = iterator.next().bounds();
+        }
+        // If we don't have it, let's instead return null island.
+        else
+        {
+            return GeoJsonUtils.geometry(GeoJsonUtils.POINT, GeoJsonUtils.coordinate(0.0, 0.0));
```

Comment:
Are you worried about a ton of output here?

Reply:
I haven't seen this condition met any any outputs thus far. This is a "just in case" situation.